python/mpt/src/mpt/hash_diff.cairo

```diff
@@ -53,3 +74,172 @@ func poseidon_storage_diff{poseidon_ptr: PoseidonBuiltin*}(diff: StorageDiffEntr
     let (storage_diff_hash) = poseidon_hash_many(5, buffer);
     return storage_diff_hash;
 }
+
+// Trie diff segment hashing
+
+// @notice Hashes a segment of account diffs into a single commitment
+// @dev All entries are hashed in order and accumulated in a buffer
+//      The buffer is then hashed to produce the final commitment
+// @param account_diff The account diff struct containing all diff entries
+// @return The hash commitment of all account diffs, or 0 if empty
+func hash_account_diff_segment{poseidon_ptr: PoseidonBuiltin*}(account_diff: AccountDiff) -> felt {
+    alloc_locals;
+    let len = account_diff.value.len;
+    if (len == 0) {
+        // TODO: Do we return 0 as the hash of an empty segment?
+        // We don't want to raise here as a block could have 0 account diffs.
+        return 0;
+    }
+    let (hashes_buffer) = alloc();
+    let buffer_len = _accumulate_diff_hashes(hashes_buffer, account_diff, 0);
+    let (final_hash) = poseidon_hash_many(buffer_len, hashes_buffer);
+    return final_hash;
+}
+
+// @notice Helper function to accumulate account diff hashes into a buffer
+// @dev Recursively processes each account diff entry and stores its hash
+// @param buffer The buffer to store the accumulated hashes
+// @param account_diff The account diff struct containing all entries
+// @param i The current index being processed
+// @return The number of hashes accumulated (buffer length)
+func _accumulate_diff_hashes{poseidon_ptr: PoseidonBuiltin*}(
+    buffer: felt*, account_diff: AccountDiff, i: felt
+) -> felt {
+    if (i == account_diff.value.len) {
+        return i;
```

Comment:
nit: is it useful to return `i` since we coerce it to be equal to account_diff.value.len?

Reply:
I left it as a "good practice" of returning the numbers of elements iterated over when constructing a memory segment, but it's not very useful